### ISSUE TITLE
Attaching screenshots after every step in Courgette report

### ISSUE THREAD

**Issue by PriyaMani328** [prashant-ramcharan/courgette-jvm#320]
Hi Prashant,

I am using cucumber BDD framework and I am using courgette plugin - 5.12.0 version as part of the framwork. I see screenshots getting attached to the courgette report if there is any failure. If I want to attach screenshots after every scenario line / after every step validation either the validation pass or failure, then is there a way to do? Please let me know. Thanks for your reply.

**Comment by prashant-ramcharan** [comment]
Hi @PriyaMani328 

You should be able to use the Cucumber `AfterStep` annotation which will run after every step.

````java
    @AfterStep
    public void afterStep(Scenario scenario) {
       // take a screenshot here
    }
````

I noticed an issue and will need to fix the Courgette html report to embed screenshots at the step level. This is why currently you don't see screenshots for every step in the Courgette html report.

However, you will still be able to see the screenshots on a step level in the Cucumber html report.

**Comment by PriyaMani328** [comment]
Hi @prashant-ramcharan 
Thanks for your reply.
I am using Courgette plugin for parallel execution and I cant call @AfterStep method as part of Courgette Test Runner.
Also, I don't have access to the Hook class as it comes from base framework. So, I cant use cucumber AfterStep method to attach the screenshots for every step.

Please let me know when is the next release where I can see screenshots for every step in the courgette report?

**Comment by prashant-ramcharan** [comment]
Hi @PriyaMani328 

Courgette will uses data provided by Cucumber so if you need to see screenshots after every step, you need to code this in your Cucumber step definition / hook class.

You dont actually call the `afterStep` method, Cucumber will automatically run the code with methods annotated with Cucumber annotations (i.e `@AfterStep`). The same way it does for before / after hooks.

_Note. You can add the `@AfterStep` annotation to any step definition class. It doesn't need to specifically be in a hook class._

**Comment by prashant-ramcharan** [comment]
@PriyaMani328 This is now released in Courgette version `6.4.1`
